### PR TITLE
LORE-95: Transformation REST API

### PR DESCRIPTION
**Pull request by figliold**
Initial Transformation REST API
@brendan-hofmann 
@mcalcote 
@josephthweatt 
@cjlange 
@mdang8

### CODE REVIEW

File: rest/specs/src/assembly/specs.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    <includeBaseDirectory>false</includeBaseDirectory>
+
+    <formats>
+        <format>zip</format>
+    </formats>
+
+    <fileSets>
+        <fileSet>
+            <directory>${project.build.directory}/classes/openapi</directory>
+            <outputDirectory></outputDirectory>
```
Comment (cjlange) [inline rest/specs/src/assembly/specs.xml]: @figliold - ❓ Is this needed if there is no value?

File: rest/specs/src/main/filtered-resources/openapi/openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+      description: The transformation request was accepted for processing.
+      headers:
+        'Content-Version':
+          $ref: '#/components/headers/ContentVersion'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/TransformResponse'
+    BadRequest:
+      description: The client message could not understood by the server due to invalid format or syntax.
```
Comment (cjlange) [inline rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: @figliold - ✏️ Minor wording fix: "...could not *be* understood by the server..."

File: rest/specs/src/main/filtered-resources/openapi/openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+          example: https://bucket.s3.amazonaws.com
+        productLocation:
+          type: string
+          description: The final location of the ingested product.
+          minLength: 1
+          maxLength: 2048
+          example: https://bucket.s3.amazonaws.com
+        mimeType:
+          type: string
+          description: The mime type of the ingest product.
```
Comment (cjlange) [inline rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: @figliold - ✏️ Should this say "...the *ingested* product" to be consistent with the other descriptions?

**Comment by brendan-hofmann** [comment]
build now

**Comment by brendan-hofmann** [comment]
build now

**Comment by brendan-hofmann** [comment]
build now

**Comment by brendan-hofmann** [comment]
build now

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+    agent {
+        node {
+            label 'linux-small'
+            customWorkspace "/jenkins/workspace/${env.JOB_NAME}/${env.BUILD_NUMBER}"
+        }
+    }
+    environment {
+        DISABLE_DOWNLOAD_PROGRESS_OPTS = '-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn ' 
+        LINUX_MVN_RANDOM = '-Djava.security.egd=file:/dev/./urandom'
+        COVERAGE_EXCLUSIONS = '**/test/**/*,**/itests/**/*,**/*Test*,**/sdk/**/*,**/*.js,**/node_modules/**/*,**/jaxb/**/*,**/wsdl/**/*,**/nces/sws/**/*,**/*.adoc,**/*.txt,**/*.xml'
```
[the remark this answers is not in the document]
Comment (figliold) [reply Jenkinsfile]: fixed

**Comment by figliold** [comment]
build now

File: .github/PULL_REQUEST_TEMPLATE [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#### What does this PR do?
+
+#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
+
```
Comment (mcalcote) [inline .github/PULL_REQUEST_TEMPLATE]: Should we add the transform team names: @cjlange, @mdang8, @josephthweatt, @brendan-hofmann, @figliold, @mcalcote

**Comment by brendan-hofmann** [comment]
Looks like we just need to run `spotless:apply` to get the build to pass.

**Comment by figliold** [comment]
build now

**Comment by figliold** [comment]
build now

**Pull request merged**